keanu-project/src/main/java/io/improbable/keanu/algorithms/NetworkSamples.java

```diff
@@ -62,6 +63,10 @@ public IntegerTensorVertexSamples getIntegerTensorSamples(VertexId vertexId) {
     }
 
     public NetworkSamples drop(int dropCount) {
+        Preconditions.checkArgument(dropCount >= 0, "Cannot drop %s samples. Drop count must be positive.", dropCount);
+        if (dropCount == 0) {
+            return this;
```

Comment:
Can we chuck a `log.warn` statement in here saying "You're trying to drop zero samples, this will do nothing"

Reply:
That's part of the contract of drop I'd say. It's not ambiguous what drop(0) means and it is valid so I don't think we should bother the user with it. I'll javadoc the method to call out that dropping 0 does nothing.